### PR TITLE
`GroupedPredictor` patch

### LINKED ISSUE #616: [BUG] Error when calling predict_proba with GroupedPredictor using shrinkage and global model
When trying to generate probabilities calling `predict_proba` from a fitted instance of `GroupedPredictor` with `shrinkage` not None and `use_global_model=True` the following error is generated:

image

Code to reproduce the error:

image

```
import sklearn
import sklego
from sklego.datasets import load_hearts
from sklearn.linear_model import LogisticRegression
from sklego.meta import GroupedPredictor

## sklego.__version__   0.7.4
## sklearn.__version__  1.0.2

df_hearts = load_hearts(as_frame=True)
model = GroupedPredictor(
    LogisticRegression(max_iter=1000), groups=['sex'],
    use_global_model=True, shrinkage="relative",
).fit(df_hearts.drop(columns=['thal','target']), df_hearts['target'])

preds = model.predict_proba(df_hearts.drop(columns=['thal','target']))
```

### PR DESCRIPTION
**Pull request by FBruzzesi**
# Description

As discussed in #616, shrinkage was intended for regression problems and even if in some cases it doesn't raise errors, it would yield incorrect results for classification or outlier detection tasks.

This PR:
- Early raises an error for shrinkage in non-regression tasks
- Expands on `GroupedPredictor` docstring
- Adds two classes, `GroupedRegressor` and `GroupedClassifier` - the latter doesn't allow for shrinkage in `__init__`
- Fixes image rendering in docs
- Removed `GroupedEstimator` as we passes 0.7.0

Fixes #616

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [X] My code follows the style guidelines (flake8)
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (also to the readme.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added tests to check whether the new feature adheres to the sklearn convention
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: sklego/meta/grouped_predictor.py [hunk 1]
```diff
@@ -409,3 +437,40 @@ def decision_function(self, X):
...
+class GroupedRegressor(GroupedPredictor):
+    def fit(self, X, y):
+        if not is_regressor(self.estimator):
+            raise ValueError("GroupedRegressor is only available for regression models")
+
+        return super().fit(X, y)
+
+
+class GroupedClassifier(GroupedPredictor):
+    def __init__(
```
Comment (FBruzzesi) [inline sklego/meta/grouped_predictor.py:456]: Forgot docstrings for these two fellas

File: sklego/meta/grouped_predictor.py [hunk 2]
```diff
@@ -409,3 +437,40 @@ def decision_function(self, X):
...
+        estimator,
+        groups,
+        use_global_model=True,
+        check_X=True,
+        **shrinkage_kwargs,
+    ):
+        super().__init__(
+            estimator=estimator,
+            groups=groups,
+            shrinkage=None,
```
Comment (FBruzzesi) [inline sklego/meta/grouped_predictor.py:467]: Forcing shrinkage to be None

File: docs/user-guide/meta-models.md [hunk 3]
```diff
@@ -58,9 +58,7 @@ Wall time: 917 ms
 
 ## Grouped Prediction
 
-<p align="center">
-  <img src="../_static/meta-models/grouped-model.png" />
-</p>
+![grouped-model](../_static/meta-models/grouped-model.png)
```
Comment (koaning) [inline docs/user-guide/meta-models.md:61]: The only real thing to comes to mind is to mention the classifier/regressor objects in the docs here maybe?

File: sklego/meta/grouped_predictor.py [hunk 4]
```diff
@@ -409,3 +437,95 @@ def decision_function(self, X):
...
+        ValueError
+            If the supplied estimator is not a regressor.
+        """
+        if not is_regressor(self.estimator):
+            raise ValueError("GroupedRegressor is only available for regression models")
+
+        return super().fit(X, y)
+
+
+class GroupedClassifier(GroupedPredictor, ClassifierMixin):
```
Comment (koaning) [inline sklego/meta/grouped_predictor.py:482]: Do we want to add tests for this?

File: sklego/meta/grouped_predictor.py [hunk 5]
```diff
@@ -409,3 +437,95 @@ def decision_function(self, X):
...
         else:
             return self.__predict_shrinkage_groups(X_group, X_value, method="decision_function")
+
+    @property
+    def _estimator_type(self):
+        """Computes `_estimator_type` dynamically from the wrapped model."""
+        return self.estimator._estimator_type
+
+
+class GroupedRegressor(GroupedPredictor, RegressorMixin):
```
Comment (koaning) [inline sklego/meta/grouped_predictor.py:447]: Do we want to add tests for this?

File: docs/user-guide/meta-models.md (hunk 3 above)
Comment (FBruzzesi) [reply docs/user-guide/meta-models.md:61]: Sounds reasonable

File: tests/test_meta/test_grouped_predictor.py [hunk 6]
```diff
@@ -632,3 +635,36 @@ def test_has_decision_function():
...
+        (GroupedRegressor, LinearRegression(), does_not_raise()),
+        (GroupedClassifier, LogisticRegression(), does_not_raise()),
+        (GroupedRegressor, LogisticRegression(), pytest.raises(ValueError)),
+        (GroupedClassifier, LinearRegression(), pytest.raises(ValueError)),
+    ],
+)
+def test_specialized_classes(meta_cls, estimator, context):
+    df = load_chicken(as_frame=True)
+    with context:
+        meta_cls(estimator=estimator, groups="diet").fit(df[["time", "diet"]], df["weight"].astype(int))
```
Comment (FBruzzesi) [inline tests/test_meta/test_grouped_predictor.py:652]: Specialized classes raise with wrong estimator type

File: sklego/meta/grouped_predictor.py [hunk 7]
```diff
@@ -409,3 +437,99 @@ def decision_function(self, X):
...
+        return super().fit(X, y)
+
+
+class GroupedClassifier(GroupedPredictor, ClassifierMixin):
+    """`GroupedClassifier` is a meta-estimator that fits a separate classifier for each group in the input data.
+
+    Its equivalent to [`GroupedPredictor`][sklego.meta.grouped_predictor.GroupedPredictor] with `shrinkage=None`
+    but it is available only for classification models.
+
+    !!! info "New in version 0.7.5"
```
Comment (FBruzzesi) [inline sklego/meta/grouped_predictor.py:490]: How do you feel about this?

Proof of rendering:
image

**Review by koaning — APPROVED** [review]
LGTM!

**Comment by koaning** [comment]
@FBruzzesi Any concerns about releasing this right away?

**Comment by FBruzzesi** [comment]
> @FBruzzesi Any concerns about releasing this right away?

Not sure if it's better to go 0.7.5 (as in the _info_ admonitions) or to 0.8.0 as it has breaking changes (no more shrinkage for classification).

If you want to wait for the `HierarchicalPredictor`s, I may need a few more days just to open the PR

**Pull request merged**